pkg/client/log/log.go

```diff
@@ -0,0 +1,617 @@
+// Copyright 2019-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package log
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/atomix/api/proto/atomix/headers"
+	api "github.com/atomix/api/proto/atomix/log"
+	"github.com/atomix/go-client/pkg/client/primitive"
+	"github.com/atomix/go-client/pkg/client/session"
+	"github.com/atomix/go-client/pkg/client/util"
+	"google.golang.org/grpc"
+)
+
+// Type is the log type
+const Type primitive.Type = "Log"
+
+// Index is the index of an entry
+type Index uint64
+
+// Version is the version of an entry
+type Version uint64
```

Comment:
We don’t need a version because entries are immutable. Versions are only needed for optimistic locking when updating objects.

Reply:
Done